server/src/main/java/org/opensearch/extensions/DiscoveryExtension.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.extensions;
+
+import org.opensearch.Version;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.cluster.node.DiscoveryNodeRole;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.transport.TransportAddress;
+import org.opensearch.common.xcontent.ToXContentFragment;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.plugins.PluginInfo;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Discover extensions running independently or in a separate process
+ *
+ * @opensearch.internal
+ */
+public class DiscoveryExtension extends DiscoveryNode implements Writeable, ToXContentFragment {
```

Comment:
Should this be called `DiscoveryExtensionNode`? 

It sounds strange either way, are there other names like this? I would expect it to be called `ExtensionNode`. The fact that it was discovered doesn't matter, doesn't it? WDYT?

Reply:
I think we should go with `DiscoveryExtensionNode`, since it's extending `DiscoveryNode` but is being tailored for extensions.